src/bots/ethfinex-badge.js

```diff
@@ -0,0 +1,45 @@
+const delay = require('delay')
+const _ethfinexBadge = require('../contracts/arbitrable-address-list.json')
+
+module.exports = async (web3, batchedSend) => {
+  // Instantiate the Arbitrable Address List contract.
+  const ethfinexBadge = new web3.eth.Contract(
+    _ethfinexBadge.abi,
+    process.env.ETHFINEX_BADGE_CONTRACT_ADDRESS
+  )
+
+  while (true) {
+    const numSubmissions = await ethfinexBadge.methods.addressCount().call()
+    const challengePeriodDuration = await ethfinexBadge.methods
+      .challengePeriodDuration()
+      .call()
+
+    // Loop over all badge submissions and timeout undisputed pending requests.
+    for (let i = 0; i < Number(numSubmissions.count); i++) {
```

Comment:
Can we loop just over pending items using the query function? Otherwise, this is going to really clog our Infura allowance.

Reply:
Switched to event logs.